repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -151,8 +151,8 @@ class Session(
           statement.compareAndTransit(StatementState.Cancelling, StatementState.Cancelled)
         } else {
           _sc.foreach(_.cancelJobGroup(statementId.toString))
+          Thread.sleep(livyConf.getTimeAsMs(RSCConf.Entry.JOB_CANCEL_TRIGGER_INTERVAL))
```

Comment:
Yes it's async but we are also calling it in a loop. Just trying to improve the best case latency here. If you feel strongly about this, I will not insist on the change.